src/backend/distributed/planner/fast_path_router_planner.c

```diff
@@ -103,15 +103,23 @@ PlannedStmt *
 GeneratePlaceHolderPlannedStmt(Query *parse)
 {
 	PlannedStmt *result = makeNode(PlannedStmt);
+#if PG_VERSION_NUM >= PG_VERSION_16
+	SeqScan *scanNode = makeNode(SeqScan);
+	Plan *plan = &(scanNode->scan.plan);
+#else
 	Scan *scanNode = makeNode(Scan);
```

Comment:
Can we make this as `SeqScan` too?

In pre-16, both are same.

`typedef Scan SeqScan;`

Reply:
```
#define get_plan_from_scan_node(node) node->scan.plan
#define get_plan_from_scan_node(node) node.plan
```

Reply:
Turns out SeqScan is already defined in all PG versions. It will get less readable to find a solution so I will just leave it as is.